lib/matplotlib/_pylab_helpers.py

```diff
@@ -47,6 +59,50 @@ def get_fig_manager(cls, num):
         return manager
 
     @classmethod
+    def show_all(cls, block=None):
+        """
+        Show all figures.  If *block* is not None, then
+        it is a boolean that overrides all other factors
+        determining whether show blocks by calling mainloop().
+        The other factors are:
+        it does not block if run inside ipython's "%pylab" mode
+        it does not block in interactive mode.
+        """
+
+        managers = cls.get_all_fig_managers()
+        if not managers:
+            return
+
+        for manager in managers:
+            manager.show()
+
+        if block is not None:
+            if block:
```

Comment:
Nono, I really dislike the innards of this method and like that @WeatherGod brought attention to it.  I did this very early on in the beginning, and got quite annoyed by this part of ``Gcf``, trying to figure out how I should refactor this part of ``Show``.  Now that the rest of the MEP feels stable (at least to me), I think I can come back and fix up this function nicely.